Sources/Toast/Toast.swift

```diff
@@ -0,0 +1,142 @@
+         - Parameter isPresented: Is the `Toast` presented or not
+         - Parameter colorProvider: ColorProvider used for styling the `Toast`, default value is read from `Config`
+         */
+        public init(
+            style: Warp.ToastStyle,
+            title: String,
+            toastEdge: Warp.ToastEdge,
+            isPresented: Binding<Bool>,
+            colorProvider: ColorProvider = Config.colorProvider
+        ) {
+            self.style = style
+            self.title = title
+            self.toastEdge = toastEdge
+            self._isPresented = isPresented
+            self.colorProvider = colorProvider
+        }
+
+        public var body: some View {
+            contentView
+            .overlay(
+                RoundedRectangle(cornerRadius: toastCornerRadius)
+                    .stroke(style.subtleBorderColor(from: colorProvider), lineWidth: 2)
+            )
+            .frame(maxWidth: .infinity)
+            .background(style.backgroundColor(from: colorProvider))
+            .cornerRadius(toastCornerRadius)
+            .transition(AnyTransition.move(edge: toastEdge.asEdge).combined(with: .opacity))
+            .onTapGesture {
+                withAnimation {
+                    isPresented.toggle()
+                }
+            }
+        }
+
+        private var contentView: some View {
+            HStack(spacing: 0) {
+                style.icon
+
+                Text(title)
+                    .padding(.leading, 8)
+                    .foregroundStyle(style.textColor(from: colorProvider))
+                    .font(from: Typography.body)
+
+                Spacer()
+
+                Image("icon_toast-close", bundle: .module)
+            }
+            .padding(16)
+        }
+    }
+}
+
+extension Warp.ToastStyle {
+    fileprivate func backgroundColor(from colorProvider: ColorProvider) -> Color {
+        switch self {
+        case .error:
+            colorProvider.toastNegativeBackground
+        case .success:
+            colorProvider.toastPositiveBackground
+        case .warning:
+            colorProvider.toastWarningBackground
+        }
+    }
+
+    fileprivate func subtleBorderColor(from colorProvider: ColorProvider) -> Color {
+        switch self {
+        case .error:
+            colorProvider.toastNegativeSubtleBorder
+        case .success:
+            colorProvider.toastPositiveSubtleBorder
+        case .warning:
+            colorProvider.toastWarningSubtleBorder
+        }
+    }
+
+    fileprivate func textColor(from colorProvider: ColorProvider) -> Color {
+        switch self {
+        case .error:
+            colorProvider.toastNegativeText
+        case .success:
+            colorProvider.toastPositiveText
+        case .warning:
+            colorProvider.toastWarningText
+        }
+    }
+
+    fileprivate var icon: Image {
+        switch self {
+        case .error:
+            Image("icon_toast-error", bundle: .module)
```

Comment:
These images need tint color from colorProvider too. As you can see in the image attached they should be different in Finn and Tori. Probably it is toast***Icon
<img width="1118" alt="Screenshot 2024-01-31 at 13 06 10" src="https://github.com/warp-ds/warp-ios/assets/89784012/8bd8fcc1-2f6d-41c9-b237-cebed00acbc0">

Reply:
The background colors do not match either, specially on Tori but it seems you have used the right colors so probably something wrong on Figma. I will investigate that.
<img width="825" alt="Screenshot 2024-01-31 at 13 06 20" src="https://github.com/warp-ds/warp-ios/assets/89784012/6142ed3e-648d-4ec3-bca6-20bb5735fe84">

Reply:
And for the icons is you have the same problem as the one in the Alert (the icon does not have transparent exclamation mark) then you can not use tint probably. Just do it like Alert (use systemIcons) until we get correct icons.

And also, the icons seems duplicate with the ones we have for Alert. So maybe we can reuse them, you can find the icons [here](https://www.figma.com/file/1qkEGDQWaftkOL7C360qVD/Fabric---Icon-library?type=design&node-id=3482-2577&mode=design&t=OAQYXYQQSne58IWo-0) and they are called Alert, so should be okey to reuse later on when we have the correct icons.

Reply:
Yes the colors on Figma is wrong and our designers are fixing that, so we can ignore that. But the icon need taking care of. And the border, it seems ours is a little bit narrow, I can see you have used 2 just like figma, but could you please make it bigger and see if it looks more like figma?

Reply:
> But the icon need taking care of.

Sure thing ✅ 

I've changed so we now use the system icons and delete my Toast variants. Then once we have the right Alert icons I can re-use those (and then maybe we need to come up with different names to avoid confusion about why we're using "Alert" icons in the "Toast" for instance).

Reply:
> And the border, it seems ours is a little bit narrow, I can see you have used 2 just like figma, but could you please make it bigger and see if it looks more like figma?

I've attached two screenshots here:

**one where I've used a `lineWidth` of 3px**

![lineWidth 3px](https://github.com/warp-ds/warp-ios/assets/117636/0ab43eda-dcb7-44f8-ac1f-8cea67ce8874)

**and one where I've used a `lineWidth` of 4px**

![lineWidth 4px](https://github.com/warp-ds/warp-ios/assets/117636/29319eab-08ed-4762-989d-02f69cba653f)

I have a hard time deciding which I prefer but I _think_ I like the 4px version best. Ask me again tomorrow and I may say 3px 😅

Reply:
I think 4 looks more like the one on Figma, so maybe the way the borderWidth is setup on Figma is different. I will bring it up with my team.

And regarding using Alert icon name in Toast component we will use whatever Figma tells us to use. We will always have Figma as source of truth.
<img width="1136" alt="Screenshot 2024-02-01 at 09 46 21" src="https://github.com/warp-ds/warp-ios/assets/89784012/4033fa45-1b97-4bc0-9550-a628b547611e">

Reply:
our designers agreed on 4 too

Reply:
> our designers agreed on 4 too

Right, 4 it is, I'll update ✅